db/commitlog/commitlog.cc

```diff
@@ -1492,7 +1492,7 @@ future<db::commitlog::segment_manager::sseg_ptr> db::commitlog::segment_manager:
                 // sync metadata (size/written)
                 co_await f.flush();
                 co_await f.close();
-                f = co_await open_file_dma(filename, flags, opt);
+                f = co_await open_file_dma(filename, flags & open_flags(~int(open_flags::create)), opt);
```

Comment:
I don't think this is needed. The manual says:

       O_CREAT
              If pathname does not exist, create it as a regular file.

Since it exists, it won't be created. We don't specify O_TRUNC, do we?

Reply:
You are right, but that means the bug is a few lines up, where we treat o_creat as if it where o_creat|o_trunc, and don't read file size in the creat case. Should not ever reach a case where it matters though, esp. if removing non-recycling case.